nodestream/databases/neo4j/query.py

```diff
@@ -12,41 +12,51 @@
 RETURN batches, committedOperations, failedOperations, errorMessages
 """
 
+NON_APOCH_COMMIT_QUERY = """
+UNWIND $param_sets AS param
+CALL apoc.cypher.doIt($batched_query, {params: param}) 
+YIELD value 
+RETURN value
+"""
 
 @dataclass(slots=True, frozen=True)
 class Query:
     query_statement: str
     parameters: Dict[str, Any]
+    apoc_iterate: bool
 
     @classmethod
-    def from_statement(cls, statement: str):
-        return cls(query_statement=statement, parameters={})
+    def from_statement(cls, statement: str, apoc_iterate: bool):
+        return cls(query_statement=statement, parameters={}, apoc_iterate=apoc_iterate)
 
     def feed_batched_query(self, batched_query: str) -> "Query":
         """Feed the results of the the query into another query that will be executed in batches."""
         return Query(
-            COMMIT_QUERY,
+            {True: COMMIT_QUERY, False: NON_APOCH_COMMIT_QUERY}[self.apoc_iterate],
             {
                 "iterate_params": self.parameters,
                 "batched_query": batched_query,
                 "iterable_query": self.query_statement,
             },
+            self.apoc_iterate
         )
 
 
 @dataclass(slots=True, frozen=True)
 class QueryBatch:
     query_statement: str
     batched_parameter_sets: List[Dict[str, Any]]
+    apoc_iterate: bool
```

Comment:
We could keep `apoc_iterate` stored here. But it means that we can't extend this. Since we already pass `apoc_iterate` to the `as_query` function, we could store the batching query instead.